bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetAlarmHandler.java

```diff
@@ -0,0 +1,249 @@
+            }
+        } catch (OWNException e) {
+            logger.debug("Exception while requesting state for channel {}: {} ", channel, e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    @Override
+    protected long getRefreshAllLastTS() {
+        return lastAllDevicesRefreshTS;
+    };
+
+    @Override
+    protected void refreshDevice(boolean refreshAll) {
+        if (refreshAll) {
+            logger.debug("--- refreshDevice() : refreshing via ALARM CENTRAL UNIT... ({})", thing.getUID());
+            try {
+                send(Alarm.requestSystemStatus());
+                lastAllDevicesRefreshTS = System.currentTimeMillis();
+            } catch (OWNException e) {
+                logger.warn("Excpetion while requesting alarm system status: {}", e.getMessage());
+            }
+        } else {
+            logger.debug("--- refreshDevice() : refreshing SINGLE... ({})", thing.getUID());
+            requestChannelState(new ChannelUID(thing.getUID(), CHANNEL_ALARM_SYSTEM_STATE));
+        }
+    }
+
+    @Override
+    protected void handleMessage(BaseOpenMessage msg) {
+        logger.debug("handleMessage({}) for thing: {}", msg, thing.getUID());
+        super.handleMessage(msg);
+        ThingTypeUID thingType = thing.getThingTypeUID();
+        if (THING_TYPE_BUS_ALARM_SYSTEM.equals(thingType)) {
+            updateCU((Alarm) msg);
+        } else {
+            updateZone((Alarm) msg);
+        }
+    }
+
+    private void updateCU(Alarm msg) {
+        WhatAlarm w = (WhatAlarm) msg.getWhat();
+        if (w == null) {
+            logger.debug("Alarm.updateCU() WHAT is null. Frame={}", msg);
+            return;
+        }
+        switch (w) {
+            case SYSTEM_ACTIVE:
+            case SYSTEM_INACTIVE:
+            case SYSTEM_MAINTENANCE:
+                updateAlarmSystemState(w);
+                break;
+            case SYSTEM_DISENGAGED:
+            case SYSTEM_ENGAGED:
+                updateAlarmSystemArmed(w);
+                break;
+            case SYSTEM_BATTERY_FAULT:
+            case SYSTEM_BATTERY_OK:
+            case SYSTEM_BATTERY_UNLOADED:
+                updateBatteryState(w);
+                break;
+            case SYSTEM_NETWORK_ERROR:
+            case SYSTEM_NETWORK_OK:
+                updateNetworkState(w);
+                break;
+            case START_PROGRAMMING:
+            case STOP_PROGRAMMING:
+            case DELAY_END:
+            case NO_CONNECTION_TO_DEVICE:
+            default:
+                logger.debug("Alarm.updateCU() Ignoring unsupported WHAT {}. Frame={}", msg.getWhat(), msg);
+        }
+    }
+
+    private void updateAlarmSystemState(WhatAlarm w) {
+        if (w == Alarm.WhatAlarm.SYSTEM_ACTIVE) {
+            updateState(CHANNEL_ALARM_SYSTEM_STATE, OnOffType.ON);
+        } else {
+            updateState(CHANNEL_ALARM_SYSTEM_STATE, OnOffType.OFF);
+        }
```

Comment:
Just a suggestion: you could do it simply in one line:
```
updateState(CHANNEL_ALARM_SYSTEM_STATE, OnOffType.from(w == Alarm.WhatAlarm.SYSTEM_ACTIVE));
```
Same in your other methods.